nni/experiment/config/base.py

```diff
@@ -47,6 +47,8 @@ def __init__(self, *, _base_path: Optional[Path] = None, **kwargs):
         They will be converted to snake_case automatically.
         If a field is missing and don't have default value, it will be set to `dataclasses.MISSING`.
         """
+        if 'basepath' in kwargs:
+            _base_path = kwargs.pop('basepath')
```

Comment:
what is basepath used for?

Reply:
what is the difference between basepath and _base_path and base_path. looks messy. could you briefly explain this pr?

Reply:
Base path is used to resolve relative paths.
The name `_base_path` is used in interface to indicate internal API; and `base_path` is used for local variables.
And there is a problem: to support snake_case (used in Python code) and camelCase (used in YAML) configuration at same time, the `Config` base class removes all underscore when loading a dict, which may incorrectly rename `_base_path` to `basepath`. Line 50 and 51 patch this problem.